qualtran/bloqs/util_bloqs.py

```diff
@@ -138,6 +139,41 @@ def wire_symbol(self, soq: 'Soquet') -> 'WireSymbol':
         return directional_text_box(' ', side=soq.reg.side)
 
 
+@frozen
+class Partition(Bloq):
+    """Partition a generic index into multiple registers.
+
+    Args:
+        n: The total bitsize of the un-partitioned register
+        regs: Registers to partition into. The `side` attribute is ignored.
+        partition: `False` means un-partition instead.
+
+    Registers:
+        x: the un-partitioned register. LEFT by default.
+        [user spec]: The registers provided by the `regs` argument. RIGHT by default.
+    """
+
+    n: int
+    regs: Tuple[Register, ...]
+    partition: bool = True
+
+    @cached_property
+    def signature(self) -> 'Signature':
+        lumped = Side.LEFT if self.partition else Side.RIGHT
+        partitioned = Side.RIGHT if self.partition else Side.LEFT
+
+        return Signature(
+            [Register('x', bitsize=self.n, side=lumped)]
+            + [attrs.evolve(reg, side=partitioned) for reg in self.regs]
+        )
+
+    def dagger(self):
+        return attrs.evolve(self, partition=not self.partition)
+
+    def __str__(self):
+        return 'Partition(...)'
```

Comment:
is there a reason why you're doing `__str__` and not (or instead of) `pretty_name` and/or `short_name`?

Reply:
because I just copy pasted the code lol. I will fix

Reply:
I just deleted, the default Partition is fine I think (...) not adding much.